server/src/main/java/org/opensearch/extensions/DiscoveryExtensionNode.java

```diff
@@ -82,6 +82,16 @@ public List<ExtensionDependency> getDependencies() {
         return dependencies;
     }
 
+    public boolean dependenciesContain(ExtensionDependency dependency) {
+        for (ExtensionDependency extensiondependency : this.dependencies) {
+            // TODO also compare version
```

Comment:
Can we add version now as a part of dependency and compare it here?

Reply:
we do not need to use the version part for this issue, and I will create issue to complete this method.

Reply:
We do have version [here](https://github.com/opensearch-project/OpenSearch/pull/5628/files#diff-c9a4accc046eb92a0e6a025970b3eacacc330f40bb55b02859df05a79710c8c8R445) though

Reply:
Yeah, you are right. I will add version part here.